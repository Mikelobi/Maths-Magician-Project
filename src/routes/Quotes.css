.quote-box {
  padding-top: 5em;
}

.inner-content {
  padding: 1em;
}

.quote {
  margin-top: 20px;
  padding-left: 40px;
  letter-spacing: 0.05em;
  font-size: 1.2em;
}

.quote-btn {
  color: #fff;
  background-color: rgb(231, 219, 46);
  padding: 0.5rem;
  border-radius: 5px;
  text-decoration: none;
  width: auto;
}
