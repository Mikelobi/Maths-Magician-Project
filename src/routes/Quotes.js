/* eslint-disable react/state-in-constructor */
import React from 'react';
import { FaQuoteLeft } from 'react-icons/fa';
import Navigation from '../components/Navigation';
import './Quotes.css';

const API = 'https://gist.githubusercontent.com/camperbot/5a022b72e96c4c9585c32bf6a75f62d9/raw/e3c6895ce42069f0ee7e991229064f167fe8ccdc/quotes.json';

class Quotes extends React.Component {
  state = {
    quotes: [
      {
        quote:
          'Life isn’t about getting and having, it’s about giving and being.',
        author: 'Kevin Kruse',
      },
    ],
    index: 0,
  };

  componentDidMount() {
    fetch(API)
      .then((res) => res.json())
      .then((res) => {
        this.setState(
          {
            quotes: res.quotes,
          },
          this.getRandomIndex,
        );
      });
  }

  getRandomIndex = () => {
    const { quotes } = this.state;
    if (quotes.length > 0) {
      const index = Math.floor(Math.random() * quotes.length);
      this.setState({
        index,
      });
    }
  };

  render() {
    const { quotes, index } = this.state;
    const quote = quotes[index];

    return (
      <>
        <Navigation />
        <h3>Click the New Quote button to generate random quotes</h3>
        <div className="quote-box">
          {quote && (
            <div className="inner-content">
              <FaQuoteLeft
                style={{
                  color: 'darkcyan',
                  fontSize: '30px',
                  marginTop: '2px',
                }}
              />

              <p className="quote">{quote.quote}</p>
              <p className="fst-italic text-end" id="author">
                {' '}
                -
                {quote.author}
              </p>
            </div>
          )}
          <button
            type="button"
            className="quote-btn"
            id="new-quote"
            onClick={this.getRandomIndex}
          >
            Get New Quote
          </button>
        </div>
      </>
    );
  }
}

export default Quotes;
