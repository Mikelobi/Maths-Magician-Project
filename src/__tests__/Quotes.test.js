import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Quotes from '../routes/Quotes';

describe('Test presence of Quotes component', () => {
  test('renders the Quotes component correctly', () => {
    const navComponent = renderer
      .create(
        <BrowserRouter>
          <Quotes />
        </BrowserRouter>,
      )
      .toJSON();
    expect(navComponent).toMatchSnapshot();
  });
});
