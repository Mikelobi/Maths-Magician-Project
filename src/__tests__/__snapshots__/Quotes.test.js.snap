// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test presence of Quotes component renders the Quotes component correctly 1`] = `
Array [
  <div
    className="header"
  >
    <div>
      <h1
        className="header-title"
      >
        Math Magicians
      </h1>
    </div>
    <nav
      className="navigation"
    >
      <li
        className="navLink"
      >
        <a
          aria-current="page"
          className="link active"
          data-testid={1}
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="navLink"
      >
        <a
          className="link"
          data-testid={2}
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="navLink"
      >
        <a
          className="link"
          data-testid={3}
          href="/quotes"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </nav>
  </div>,
  <h3>
    Click the New Quote button to generate random quotes
  </h3>,
  <div
    className="quote-box"
  >
    <div
      className="inner-content"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "darkcyan",
            "fontSize": "30px",
            "marginTop": "2px",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 256h-80v-64c0-35.3 28.7-64 64-64h8c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24h-8c-88.4 0-160 71.6-160 160v240c0 26.5 21.5 48 48 48h128c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48zm-288 0H96v-64c0-35.3 28.7-64 64-64h8c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24h-8C71.6 32 0 103.6 0 192v240c0 26.5 21.5 48 48 48h128c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48z"
        />
      </svg>
      <p
        className="quote"
      >
        Life isn’t about getting and having, it’s about giving and being.
      </p>
      <p
        className="fst-italic text-end"
        id="author"
      >
         
        -
        Kevin Kruse
      </p>
    </div>
    <button
      className="quote-btn"
      id="new-quote"
      onClick={[Function]}
      type="button"
    >
      Get New Quote
    </button>
  </div>,
]
`;
